src/Containers/ActionsField/ActionsField.tsx

```diff
@@ -17,33 +17,45 @@ const ActionsField:React.FC<Props> = ({ formState, setAction }) => {
   const handleChange = (e:React.MouseEvent<HTMLElement>):void => {
     const currentTarget = e.target as HTMLElement;
 
-    let newAction = actionsCollection.filter((action:Action) => { //eslint-disable-line
-        if (action.action === currentTarget.innerText) {
-          return action
-        }
+    let newAction:(Action | undefined) = actionsCollection.find((action:Action) => {
+      if (action.action === currentTarget.innerText) {
+        return action;
+      }
+      return undefined;
     });
 
-    setSelectedAction(newAction[0]);
+    if (!newAction) {
+      newAction = {id: 1, action: 'Create an app'};
+    };
+
+    setSelectedAction(newAction);
     setIsClicked(false);
   }
 
   const toggleBlock = (): void => setIsClicked(!isClicked);
 
   const updateAction = () => {
-    let newChosenAction = actionsCollection.filter((action:Action) => { //eslint-disable-line
+    let newChosenAction: (Action | undefined) = actionsCollection.find((action:Action) => {
       if (action.action === selectedAction.action) {
-        return action
+        return action;
       }
-    })
-    setAction({...formState, action: newChosenAction[0]})
+      return undefined;
+    });
+
+    if (!newChosenAction) {
+      newChosenAction = {id: 1, action: 'Create an app'};
+    };
+
+    setAction({...formState, action: newChosenAction});
```

Comment:
Can we make it shorter? Such as:
```
if (newChosenAction) {
     setAction({...formState, action: newChosenAction});
};
```
Because we already have default value of action declared at line 13.

Reply:
I know these are redundant, but this is what I had to implement to resolve compiling issues with TypeScript. Because there is a possibility of undefined being returned, I had to add the if blocks for those edge cases to be accounted for. I suppose that we could say that the return value could be :any but that seems about as effective as using the disable eslint comment.

Maybe I am just not understanding the TypeScript convention when dealing with .find, .filter methods

Reply:
Yes, there is can be undefined result from .find method. Your if block reassigns undefined value to default and then set up a new action state.  In a short version, the action state is reassigned only when we got something from .find, and stays default when we got undefined (because we do nothing in this case).